src/simplewallet/simplewallet.cpp

```diff
@@ -3381,6 +3451,129 @@ bool simple_wallet::check_tx_key(const std::vector<std::string> &args_)
   return true;
 }
 //----------------------------------------------------------------------------------------------------
+bool simple_wallet::check_tx_proof(const std::vector<std::string> &args)
+{
+  if(args.size() != 3) {
+    fail_msg_writer() << tr("usage: check_tx_proof <txid> <address> <signature>");
+    return true;
+  }
+
+  if (!try_connect_to_daemon())
+    return true;
+
+  // parse txid
+  assert(m_wallet);
+  cryptonote::blobdata txid_data;
+  if(!epee::string_tools::parse_hexstr_to_binbuff(args[0], txid_data) || txid_data.size() != sizeof(crypto::hash))
+  {
+    fail_msg_writer() << tr("failed to parse txid");
+    return true;
+  }
+  crypto::hash txid = *reinterpret_cast<const crypto::hash*>(txid_data.data());
+
+  LOCK_IDLE_SCOPE();
```

Comment:
This was also copied from `check_tx_key()`. Where exactly should it be moved to? I have only a vague understanding that `LOCK_IDLE_SCOPE();` is used when the program accesses some data stored in the wallet from various threads concurrently.

Actually this function (and also `check_tx_key()`) should be completely independent of the wallet data (i.e. they could be implemented in the daemon), so this `LOCK_IDLE_SCOPE();` as well as the above `assert(m_wallet);` could be dropped altogether. Am I correct?